Description: Upstream changes introduced in version 0.6-3
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 python-docutils (0.6-3) unstable; urgency=low
 .
   * Update patch fixing insecure use of temporary files to be compatible with
     XEmacs 21.
   * Add ‘Depends: ${misc:Depends}’.
   * Point to versioned symlink in /usr/share/common-licenses for the GPL
     license.
   * Fix the docutils.utils.decode_path() function so that it accepts None as
     an argument (closes: #563087).
   * Force upgrade of python-epydoc affected by bug #561793.
   * Add preinst maintainer script to remove python-central leftovers
     (closes: #563696).
   * Upstream author of IE PNG Alpha Fix agreed that every version of his script
     can be distributed under the terms of the LGPL 2.1 or later version.
     Update debian/*.copyright and copies of iepngfix.htc accordingly
     (closes: #563704).
   * Remove superfluous ‘Provides: ${python:Provides}’ and ‘XB-Python-Version’
     from debian/control.
   * Conflict with utterly broken version of zope3, which infringes
     python-docutils namespace.
 .
 The person named in the Author field signed this changelog entry.
Author: Jakub Wilk <ubanus@users.sf.net>
Bug-Debian: http://bugs.debian.org/563087
Bug-Debian: http://bugs.debian.org/563696
Bug-Debian: http://bugs.debian.org/563704

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- python-docutils-0.6.orig/tools/editors/emacs/rst.el
+++ python-docutils-0.6/tools/editors/emacs/rst.el
@@ -3304,65 +3304,17 @@ of the entire buffer, if the region is n
 (defvar rst-temp-dir nil)
 (make-variable-buffer-local 'rst-temp-dir)
 
-;; make-temp-file is not available in XEmacs 21
-(if (fboundp 'make-temp-file)
-  (defun rst-make-temp-dir (prefix) (make-temp-file prefix t))
-  (defun rst-make-temp-dir (prefix)
-    (let ((umask (default-file-modes)) file)
-      (unwind-protect
-        (progn
-          (set-default-file-modes 448) ; o700
-          (while
-            (condition-case ()
-              (progn
-                (setq file
-                  (make-temp-name
-                    (if (zerop (length prefix))
-                      (file-name-as-directory (temp-directory))
-                      (expand-file-name prefix (temp-directory))
-                    )
-                  )
-                )
-                (make-directory file)
-                nil
-              )
-              (file-already-exists t)
-            )
-            nil
-          )
-          file
-        )
-        (set-default-file-modes umask)
-      )
-    )
-  )
-)
-
 (defun rst-get-temp-dir ()
   (or rst-temp-dir
-    (setq rst-temp-dir
-      (file-name-as-directory (rst-make-temp-dir "rst-"))
-    )
-  )
-)
-
-;; dired-delete-file is not available in XEmacs 21
-(defun rst-delete-file (file)
-  (if (not (eq t (car (file-attributes file))))
-    (delete-file file)
-    (when
-      (setq files (directory-files file t "^\\([^.]\\|\\.\\([^.]\\|\\..\\)\\).*"))
-      (while files
-        (rst-delete-file (car files))
-        (setq files (cdr files))
-      )
+    (setq rst-temp-dir 
+      (file-name-as-directory (make-temp-file "rst-" t))
     )
-    (delete-directory file)
   )
 )
 
 (defun rst-remove-temp-dir ()
-  (if rst-temp-dir (rst-delete-file rst-temp-dir))
+  (require 'dired)
+  (if rst-temp-dir (dired-delete-file rst-temp-dir 'always))
 )
 
 (add-hook 'kill-buffer-hook 'rst-remove-temp-dir)
